controllers/cache_controller.go

```diff
@@ -274,50 +273,13 @@ func (r *cacheRequest) ispnCreateOrUpdate() (*ctrl.Result, error) {
 		return &ctrl.Result{}, err
 	}
 
-	if r.infinispan.IsDataGrid() {
```

Comment:
I think there's a small window when a user upgrades to the latest Operator where we can have WellFormed=True with the cache service, as the cache_controller and infinispan_controller are independent. We can guard against this by adding a check after the `if !infinispan.IsWellFormed() {` block on L190, something like:

```yaml
if infinispan.IsCache() {
   log.info("Ignoring Cache CR as cluster is marked as CacheService")
   return
}
```